# Week of 2023-09-11 to 2023-09-18 (30)

### GHFirst (10)

- [Revert "Remove c10::either (#109299)"](https://github.com/pytorch/pytorch/commit/71420a98ab9af5f7cafc57eb7cb0e3c80489771c)
  - sorry but there are a few internal usages and when I tried swapping them out, I got some errors.  I will get someone to look at them on Monday ([comment](https://github.com/pytorch/pytorch/pull/109299#issuecomment-1722579387))
- [Revert "replace torch::make_unique with std::make_unique (#108866)"](https://github.com/pytorch/pytorch/commit/525e4f42d093af4557dda6555781c1d83239527e)
  - Sorry but I found more usages of `torch::make_unique` internally, I can go change all of these, but I'd prefer if that gets done before this gets merged ([comment](https://github.com/pytorch/pytorch/pull/108866#issuecomment-1722577925))
- [Revert "[HigherOrderOp] Should automatically pop modes (#109157)"](https://github.com/pytorch/pytorch/commit/07f2efa285ceac22e560c124d09ebb3a6a8e7016)
  - broke internal builds D49346922 ([comment](https://github.com/pytorch/pytorch/pull/109157#issuecomment-1722571262))
- [Revert "python functionalization: add helpers, functionalize_sync and mirror_autograd_meta (#107917)"](https://github.com/pytorch/pytorch/commit/49b18ae5464b00b9849761afdab900a017de0652)
  - breaking internal builds D49346637 ([comment](https://github.com/pytorch/pytorch/pull/107917#issuecomment-1722566885))
- [Revert "Created nested utils.cpp (#109304)"](https://github.com/pytorch/pytorch/commit/17193faf1a05f8ebca27922a136c42e2712892f1)
  - sorry but this is breaking internal builds due to the new header file D49346814 ([comment](https://github.com/pytorch/pytorch/pull/109304#issuecomment-1722561958))
- [Revert "[inductor][Optimus]Improve logging for group batch fusion (#109314)"](https://github.com/pytorch/pytorch/commit/7af792ab05076bf68107eff87eb88d627d0c68f8)
  - Diff reverted internally ([comment](https://github.com/pytorch/pytorch/pull/109314#issuecomment-1722015037))
- [Revert "split by tag (#108892)"](https://github.com/pytorch/pytorch/commit/bf5622e9650c8dab77ccabc2fffe600bbdfc0b5b)
  - Diff reverted internally ([comment](https://github.com/pytorch/pytorch/pull/108892#issuecomment-1720249148))
- [Revert "[dynamo] preserve some FX node metadata of GraphModules (#107067)"](https://github.com/pytorch/pytorch/commit/c5e75886138ccf05c4b35ed9efd88d3b25588dc8)
  - Break internal build ([comment](https://github.com/pytorch/pytorch/pull/107067#issuecomment-1717321061))
- [Revert "Force synced KJT to trace unbacked SymInt (#108960)"](https://github.com/pytorch/pytorch/commit/1d32c9c7f236c8a3836fdf677be853c5485efbc1)
  - Diff reverted internally ([comment](https://github.com/pytorch/pytorch/pull/108960#issuecomment-1715850779))
- [Revert "[ROCm] Add ROCm AMDGPU support for inductor cpp codegen (#105141)"](https://github.com/pytorch/pytorch/commit/5a7c008b300f49b61644576f2d8e87fdd59bf210)
  - Break internal build ([comment](https://github.com/pytorch/pytorch/pull/105141#issuecomment-1715629007))

### Ignored Signal (1)

- [Revert "make sure all torch._numpy tests run on CI (#108762)"](https://github.com/pytorch/pytorch/commit/090fe45e1c35922877774893af74c835ef5e2ca9)
  - sorry but I think the asan test_scalarmath failure is real https://hud.pytorch.org/pytorch/pytorch/commit/7abeb92796635bd3ee216a0872bddd0395e97d10 https://github.com/pytorch/pytorch/actions/runs/6132913963/job/16645381921 ([comment](https://github.com/pytorch/pytorch/pull/108762#issuecomment-1714214523))

### Landrace (5)

- [Revert "[inductor] Add ir.Scan and lower aten.cumsum on CUDA (#106581)"](https://github.com/pytorch/pytorch/commit/800c66561872f1f66fc0850696fe231ef71401e4)
  - This combined with #108803 uncovered a triton bug openai/triton#2298 ([comment](https://github.com/pytorch/pytorch/pull/106581#issuecomment-1719811113))
- [Revert "Support SymBool input to torch.compile (#107850)"](https://github.com/pytorch/pytorch/commit/47f79e9a2b579ffece8367d844ce36aba602a00d)
  - Sorry for reverting this, but test_export_with_symbool_inputs is failing in trunk https://hud.pytorch.org/pytorch/pytorch/commit/a08e1370ef8cb13cfbf18d9663427a57fa8657f2 ([comment](https://github.com/pytorch/pytorch/pull/107850#issuecomment-1718675877))
- [Revert "Remove expected test failures for cond (#108709)"](https://github.com/pytorch/pytorch/commit/de76c88d90b8005f91ebc58d6a04645754fae9a5)
  - Sorry for reverting this, but test_export_with_symbool_inputs is failing in trunk https://hud.pytorch.org/pytorch/pytorch/commit/a08e1370ef8cb13cfbf18d9663427a57fa8657f2 ([comment](https://github.com/pytorch/pytorch/pull/108709#issuecomment-1718669964))
- [Revert "Reland 2: Add PyObject preservation for UntypedStorage (#109039)"](https://github.com/pytorch/pytorch/commit/59f605be577054547dc569cd373e6341b5b9c9ea)
  - Sorry for reverting your change but it is failing linter job in trunk, probably due to a landrace ([comment](https://github.com/pytorch/pytorch/pull/109039#issuecomment-1715147020))
- [Revert "Add Opinfos for the Tensor overload of linspace/logspace (#107958)"](https://github.com/pytorch/pytorch/commit/e276d70451efdbdff599ce97e986c447063dc3fb)
  - I think the newly added test test_mps.py::TestConsistencyCPU::test_output_match_logspace_tensor_overload_cpu_complex64 is broken, probably a landrace since the mergebase seems to be 21 days old https://hud.pytorch.org/pytorch/pytorch/commit/106e0a0ef19c8dad088fc1ec10d7d93d76409352 https://github.com/pytorch/pytorch/actions/runs/6149523234/job/16685849126 ([comment](https://github.com/pytorch/pytorch/pull/107958#issuecomment-1714309905))

### Not through pytorchbot (5)

- [Revert "[PyTorch] Add Expanded call stack to nodes (#108426)" (#109468)](https://github.com/pytorch/pytorch/commit/0cae3b5df52a2c614ad8f3939543d9057fff80a9)
- [Back out "Revert D49107540: [pytorch][PR] split by tag" (#109332)](https://github.com/pytorch/pytorch/commit/393fe9339a162b02c546ebe6cea1bb87d403d0f8)
- [Back out "[Inductor] Extend Pattern Matcher to Match Equivalent Function Invocation (#107832)" (#109174)](https://github.com/pytorch/pytorch/commit/8851603a9c504d1a11045829ca26eae0cbb24225)
- [Back out "Dynamo support for autograd.Function w/ once_differentiable (#108686)" (#109199)](https://github.com/pytorch/pytorch/commit/f2639a2c3719141f3dcf4dd937ccba64c0259d25)
- [Revert "CI Sev - pin docker images for A100 workers (#108871)" (#109071)](https://github.com/pytorch/pytorch/commit/15b13d3cff1c0635bae95983b7cbb17684a28a76)

### No Signal (6)

- [Revert "Add meta and OpInfo for _embedding_bag_dense_backward (#109211)"](https://github.com/pytorch/pytorch/commit/be9f73f0314252475b38c05d4337a99450c8482e)
  - Sorry I think the test_ops.py::TestCommonCUDA::test_compare_cpu__embedding_bag_dense_backward_cuda_float32 is failing https://hud.pytorch.org/pytorch/pytorch/commit/492a93d185a74bb4ccb9dea184a71f43f8952083 https://github.com/pytorch/pytorch/actions/runs/6190707847/job/16808644559 not sure why this is run in slow when it looks to be a new test ([comment](https://github.com/pytorch/pytorch/pull/109211#issuecomment-1720235918))
- [Revert "add Half support for BatchNorm on CPU (#102070)"](https://github.com/pytorch/pytorch/commit/b226373d1607b157640276e86277e9b966d420e3)
  - I'm very sorry but it looks like #106543 was not fixed, I still see it failing on main https://hud.pytorch.org/pytorch/pytorch/commit/b6a1d3fb97ca8eeccf15a4c495fdd1af4b197f88 https://github.com/pytorch/pytorch/actions/runs/6185704949/job/16793975677 ([comment](https://github.com/pytorch/pytorch/pull/102070#issuecomment-1719747065))
- [Revert "add Half support for BatchNorm on CPU (#102070)"](https://github.com/pytorch/pytorch/commit/04a765f95d96a19a9ce4e214a3f58e105f09b5d0)
  - sorry it looks like this is causing an unexpected success for `test_jit_fuser_te.py::TestNNCOpInfoCPU::test_nnc_correctness_nn_functional_batch_norm_cpu_float16` https://hud.pytorch.org/pytorch/pytorch/commit/6065e7a97cfad4c2ae2b8722969648a53265fa13 https://github.com/pytorch/pytorch/actions/runs/6178069462/job/16770849782 ([comment](https://github.com/pytorch/pytorch/pull/102070#issuecomment-1718402208))
- [Revert "[inductor] Lower masked_scatter on CUDA (#108803)"](https://github.com/pytorch/pytorch/commit/91aab161d05415c9c7f4519a31eb8a14018d94cd)
  - makes test_comprehensive_masked_scatter_cuda_int64 flaky ([comment](https://github.com/pytorch/pytorch/pull/108803#issuecomment-1716407433))
- [Revert "[inductor] Parallelize Max Autotune step 1: Use Popen (#107982)"](https://github.com/pytorch/pytorch/commit/2039f30c062ec225bf1add9a887c7d8b30a8b6a9)
  - fbcode failures ([comment](https://github.com/pytorch/pytorch/pull/107982#issuecomment-1714818307))
- [Revert "[inductor] Parallelize Max Autotune step 2: Use all GPUs (#107983)"](https://github.com/pytorch/pytorch/commit/c36c2bfcb2c650488ac866cbdaf2bd717b89b5b1)
  - fbcode failures ([comment](https://github.com/pytorch/pytorch/pull/107983#issuecomment-1714816358))

### Weird (3)

- [Revert "[inductor] Lower masked_scatter on CUDA (#108803)"](https://github.com/pytorch/pytorch/commit/900288f138529e5e8ed83ebda2c5b782751c9b30)
  - Bot merged after aborted rebase ([comment](https://github.com/pytorch/pytorch/pull/108803#issuecomment-1719918831))
- [Revert "Remove fixed skips (#108674)"](https://github.com/pytorch/pytorch/commit/41bd0fde7e886d7b08097e9a1123baa08e18c929)
  - Sorry for picking this up a bit late, but with https://github.com/pytorch/pytorch/pull/108647 reverted, these tests are failing again. So we need to wait for the PR to reland before we can land this change ([comment](https://github.com/pytorch/pytorch/pull/108674#issuecomment-1715202692))
- [Revert "Introduce Tensor overload to linspace and logspace (#104889)"](https://github.com/pytorch/pytorch/commit/a7f5abeade9200358b57c2bde6c0efcad56904f6)
  - sorry have to revert this to revert https://github.com/pytorch/pytorch/pull/107958 ([comment](https://github.com/pytorch/pytorch/pull/104889#issuecomment-1714305768))
